axum-macros/src/typed_path.rs

```diff
@@ -18,52 +18,80 @@ pub(crate) fn expand(item_struct: ItemStruct) -> syn::Result<TokenStream> {
         ));
     }
 
-    let Attrs { path } = parse_attrs(attrs)?;
+    let Attrs { path, rejection } = parse_attrs(attrs)?;
 
     match fields {
         syn::Fields::Named(_) => {
             let segments = parse_path(&path)?;
-            Ok(expand_named_fields(ident, path, &segments))
+            Ok(expand_named_fields(ident, path, &segments, rejection))
         }
         syn::Fields::Unnamed(fields) => {
             let segments = parse_path(&path)?;
-            expand_unnamed_fields(fields, ident, path, &segments)
+            expand_unnamed_fields(fields, ident, path, &segments, rejection)
         }
-        syn::Fields::Unit => expand_unit_fields(ident, path),
+        syn::Fields::Unit => expand_unit_fields(ident, path, rejection),
     }
 }
 
+mod kw {
+    syn::custom_keyword!(rejection);
+}
+
 struct Attrs {
     path: LitStr,
+    rejection: Option<syn::Path>,
+}
+
+impl Parse for Attrs {
+    fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
+        let path = input.parse()?;
+
+        let _ = input.parse::<Token![,]>();
+
+        let lh = input.lookahead1();
+        let rejection = if lh.peek(kw::rejection) {
+            input.parse::<kw::rejection>()?;
+            let content;
+            syn::parenthesized!(content in input);
+            Some(content.parse()?)
+        } else {
+            None
+        };
```

Comment:
```suggestion
        } else if lh.is_empty() {
            None
        } else {
            return Err(lh.error());
        };
```

Reply:
But `rejection(...)` is optional.

Reply:
That's what the `is_empty()` branch is for: If there is no further input after the (optional) comma (by the way: it will probably work to omit is even when `rejection(...)` follows), rejection will be `None`.

Reply:
I guess the most correct solution would be to check `input.is_empty()` before attempting to parse the comma, and if the input isn't empty, just parse a comma plus the `rejection` stuff without lookahead, but also without any `is_ok()` or other checks (instead just `?`s to error out if something unexpected follows).

Reply:
ah of course yes! I somehow missed the empty branch.